elements/rh-navigation-primary/rh-navigation-primary-item.ts

```diff
@@ -0,0 +1,126 @@
+import type { IconNameFor, IconSetName } from '@rhds/icons';
+
+import { LitElement, html, isServer } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { state } from 'lit/decorators/state.js';
+import { query } from 'lit/decorators/query.js';
+import { property } from 'lit/decorators/property.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+import { consume } from '@lit/context';
+import { context } from './context.js';
+
+import { InternalsController } from '@patternfly/pfe-core/controllers/internals-controller.js';
+
+import '@rhds/elements/rh-icon/rh-icon.js';
+
+import './rh-navigation-primary-item-menu.js';
+
+import styles from './rh-navigation-primary-item.css';
+
+@customElement('rh-navigation-primary-item')
+export class RhNavigationPrimaryItem extends LitElement {
+  static readonly styles = [styles];
+
+  #highlight = false;
+
+  // eslint-disable-next-line no-unused-private-class-members
+  #internals = InternalsController.of(this, { role: 'listitem' });
+
+  #hydrated = false;
+
+  @query('details')
+  private _details!: HTMLDetailsElement;
+
+  @property({ type: Boolean, reflect: true }) open = false;
+
+  /* Summary text for dropdown variants only */
+  @property() summary?: string;
+
+  /* Variants 'link' | 'dropdown', link is the default if no variant is given */
+  @property() variant?: 'link' | 'dropdown' = 'link';
+
+  /**
+   * Hides the element at various container query based breakpoints.
+   * Breakpoints available 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl'
+   */
+  @property({ reflect: true, attribute: 'hide-at' })
+  hideAt?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' = undefined;
+
+  /** Shorthand for the `icon` slot, the value is icon name */
+  @property() icon?: IconNameFor<IconSetName>;
+
+  /** Icon set for the `icon` property - 'ui' by default */
+  @property({ attribute: 'icon-set' }) iconSet?: IconSetName;
+
+  @consume({ context, subscribe: true })
+  @state()
+  private compact?: boolean;
+
+
+  /**
+   * Color palette
+   */
+  @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette;
```

Comment:
keep themable, but remove this

Reply:
Ok, yeah, that makes sense, given this is a lightdom element. A primary nav item may be the first to instantiate on the page and provide the theme styles.  Good catch.